### ISSUE TITLE
'did-finish-load' event sets window readyState from READY back to LOADING

### ISSUE THREAD

**Issue by nidefawl** [microsoft/vscode#140762]
Does this issue occur when all extensions are disabled?: I modified vs\code\electron-browser\workbench\workbench.html

- VS Code Version: I tested it on ea2afbaf90cb6d9672d8a681e90492d23c687064
- OS Version: Win10 21H2

Steps to Reproduce:

1. Add slow loading resource to workbench.html
```html
	<!-- Startup via workbench.js -->
	<script src="workbench.js"></script>
	<img src="https://i.imgur.com/1FGwV3X.jpeg" />
```
2. In devtools disable cache and enable network throttling. Slow 3G should work to make sure the loading completes after window is in ready state.

A reload fires events in this order:

> [main 2022-01-15T05:24:15.509Z] window (main): ReadyState.NAVIGATING
> [main 2022-01-15T05:24:19.242Z] window (main): ReadyState.READY
> [main 2022-01-15T05:24:20.797Z] window (main): did-finish-load event
> [main 2022-01-15T05:24:20.797Z] window (main): ReadyState.LOADING

That leaves the window in ReadyState.LOADING

**Comment by bpasero** [comment]
I tried to reproduce with a different image:

```html
<!-- Copyright (C) Microsoft Corporation. All rights reserved. -->
<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8" />
		<meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src 'self' https: data: blob: vscode-remote-resource:; media-src 'self'; frame-src 'self' vscode-webview:; object-src 'self'; script-src 'self' 'unsafe-eval' blob:; style-src 'self' 'unsafe-inline'; connect-src 'self' https: ws:; font-src 'self' https: vscode-remote-resource:;">
		<meta http-equiv="Content-Security-Policy" content="require-trusted-types-for 'script'; trusted-types amdLoader cellRendererEditorText defaultWorkerFactory diffEditorWidget editorGhostText domLineBreaksComputer editorViewLayer diffReview dompurify notebookRenderer safeInnerHtml standaloneColorizer tokenizeToString;">
	</head>
	<body aria-label="">
	</body>

	<!-- Init Bootstrap Helpers -->
	<script src="../../../../bootstrap.js"></script>
	<script src="../../../../vs/loader.js"></script>
	<script src="../../../../bootstrap-window.js"></script>

	<!-- Startup via workbench.js -->
	<script src="workbench.js"></script>
	<img src ="https://www.kenrockwell.com/nikon/d810/sample-images/file-types/810_3433-LARGE-BASIC-OPT-QUAL.JPG" />
</html>
```

and was not able to get into that state.

**Comment by bpasero** [comment]
But I do acknowledge a theoretical problem here: the `READY` signal is emitted from the workbench and I assume that if something loads slowly and we get the `did-finish-load` event afterwards because something is loading besides the workbench, we could get into this state.

However, I see no way of running into this issue with the `workbench.html` we have, because `workbench.js` is the last thing we load. 

How does this happen, what is the scenario?

**Comment by bpasero** [comment]
Looking closer, I think these states are a bit bogus, https://github.com/microsoft/vscode/pull/140846 attempts to clean that up and remove the one that depends on the underlying loading event. Maybe you want to review?

**Comment by nidefawl** [comment]
I don't think this PR will fix the problem. The did-finish-load event handler never checks the current value of window readyState. 

I fixed it by adding a notification for LifecyclePhase.Starting on the renderer, identical to the LifecyclePhase.Ready notification behavior that ends up calling windows setReady().

I can make PR for reference, but I have to find a better repro for the issue first.

**Comment by bpasero** [comment]
But with my change, the `did-finish-load` handler no longer changes the `ReadyState` of the window?

**Comment by nidefawl** [comment]
I just saw that. Nevermind, that means its fixed 👍
